sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/design/TimeSeriesHierarchies_API_Design.md

```diff
@@ -0,0 +1,175 @@
+    IEnumerable<string> timeSeriesHierarchyIds, 
+    CancellationToken cancellationToken = default)
+```
+
+```csharp
+/// <summary>
+/// Gets time series insight hierarchies by Time Series hierarchy Names asynchronously.
+/// </summary>
+/// <param name="timeSeriesHierarchyNames">List of names of the Time Series hierarchies to return.</param>
+/// <param name="cancellationToken">The cancellation token.</param>
+/// <returns>List of hierarchy or error objects corresponding by position to the array in the request. Hierarchy object is set when operation is successful and error object is set when operation is unsuccessful.
+public virtual async Task<Response<TimeSeriesHierarchyOperationResult[]>> GetTimeSeriesHierarchiesByNameAsync(
+    IEnumerable<string> timeSeriesHierarchyNames, 
+    CancellationToken cancellationToken = default)
+```
+
+### Create OR Replace Time Series Insights Hierarchies
+
+```csharp
+/// <summary>
+/// Creates or Updates Time Series insight hierarchies asynchronously.
+/// </summary>
+/// <param name="timeSeriesHierarchies">The Time Series insight hierarchies to be created or replaced.</param>
+/// <param name="cancellationToken">The cancellation token.</param>
+/// <returns>
+/// List of hierarchy or error objects corresponding by position to the array in the request. Hierarchy object is set when operation is successful and error object is set when operation is unsuccessful.
+/// </returns>
+public virtual async Task<Response<TimeSeriesHierarchyOperationResult[]>> CreateOrReplaceTimeSeriesHierarchiesAsync(
+    IEnumerable<TimeSeriesHierarchy> timeSeriesHierarchies,
+    CancellationToken cancellationToken = default)
+```
+
+### Delete Time Series Insights Hierarchies
+
+```csharp
+/// <summary>
+/// Deletes Time Series insight hierarchies by Time Series hierarchy Ids asynchronously.
+/// </summary>
+/// <param name="timeSeriesHierarchyIds">List of names of the Time Series hierarchies to return.</param>
+/// <param name="cancellationToken">The cancellation token.</param>
+/// <returns>
+/// List of error objects corresponding by position to the "delete" array in the request - null when the operation is successful.
+/// </returns>
+public virtual async Task<Response<TimeSeriesOperationError[]>> DeleteTimeSeriesHierarchiesByIdAsync(
+    IEnumerable<string> timeSeriesHierarchyIds,
+    CancellationToken cancellationToken = default)
+```
+
+```csharp
+/// <summary>
+/// Deletes time series insight hierarchies by Time Series hierarchy Names asynchronously.
+/// </summary>
+/// <param name="timeSeriesHierarchyNames">List of names of the Time Series hierarchies to return.</param>
+/// <param name="cancellationToken">The cancellation token.</param>
+/// <returns>
+/// List of error objects corresponding by position to the "delete" array in the request - null when the operation is successful.
+/// </returns>
+public virtual async Task<Response<TimeSeriesOperationError[]>> DeleteTimeSeriesHierarchiesByNameAsync(
```

Comment:
Don't have a good reason whether should be 'Name' or 'Names', given that it takes in a list of names. Interested to know what the team thinks. 

If you force me to choose, I vote for Names.

Reply:
Ok, I may have misheard you in the meeting yesterday to vote for otherwise. I don't have a strong preference here, I will change it.

Reply:
@drwill-ms Do you have anything to weigh in for singular "Name" vs plural "Names"?

Reply:
I'm not sure. I could see it either way. Probably better to get advice from Krzysztof because I'm guessing we have a standard there.